File: 0833-51-2594-minimum-time-to-repair-cars/0833-51-2594-minimum-time-to-repair-cars.java
class Solution {
    public boolean ok(long md,int [] ranks,long cars){
        long cnt = 0;
        for(int i = 0 ; i < ranks.length;i++){
            cars-=(long)Math.sqrt(md/(long)ranks[i]);
        }
        return cars <=0;
    }
    public long repairCars(int[] ranks, int cars) {
        long st = 1, ed = 100L*cars*cars,md,cur = -1;
        while(st<=ed){
            md = (st+ed)/2;
            if(ok(md,ranks,cars)){
                ed = md -1;
                cur = md;
            }
            else st = md+1;
        }
        return cur;
    }
}

File: 0833-51-2594-minimum-time-to-repair-cars/0833-51-2594-minimum-time-to-repair-cars.js
/**
 * @param {number[]} ranks
 * @param {number} cars
 * @return {number}
 */

var ok = function(md,ranks,cars){
    for (let r of ranks){
        cars-=Math.floor(Math.sqrt(md/r));
    }
    return cars<=0;
}
var repairCars = function(ranks, cars) {
    let st = 1;
    let ed = Math.min(...ranks) * cars * cars;
    let cur = -1;
    while(st<=ed){
        let md = Math.floor((st+ed)/2);
        if(ok(md,ranks,cars)){
            cur = md;
            ed = md-1;
        }
        else st = md+1;
    }
    return cur;
};